### PR TITLE
[DOCS] Add 7.12 release highlights

### PR DESCRIPTION
**Pull request by jrodewig**
Adds release highlights for:

* Frozen tier (#68605) and shared snapshot cache (#68714, #68509)
* Repository test kit/analyzer #69050

### Preview
https://elasticsearch_70703.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.12/release-highlights.html#frozen-tier-shared-snapshot-cache

### CODE REVIEW

**Review by ywelsch — APPROVED** [review]
@jrodewig Thank you for adding these. LGTM

File: docs/reference/release-notes/highlights.asciidoc [hunk 1]
```diff
@@ -24,6 +24,56 @@ Other versions:
...
+To add a node to the frozen tier for an on-premise cluster, assign it the
+{ref}/modules-node.html#data-frozen-node[`data_frozen`] node role and set
+{ref}/searchable-snapshots.html#searchable-snapshots-shared-cache[`xpack.searchable.snapshot.shared_cache.size`]
+in `elasticsearch.yml`. See the {ref}/data-tiers.html#frozen-tier[Data tiers]
+and
+{ref}/searchable-snapshots.html#searchable-snapshot-mount-storage-options[{search-snap}]
+documentation.
+
+[discrete]
+[[test-snapshot-repositories]]
```
Comment (DaveCTurner) [inline docs/reference/release-notes/highlights.asciidoc:63]:
```suggestion
[[analyze-snapshot-repositories]]
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 2]
```diff
@@ -24,6 +24,56 @@ Other versions:
...
+=== Test snapshot repositories
```
Comment (DaveCTurner) [inline docs/reference/release-notes/highlights.asciidoc:64]:
```suggestion
=== Analyze snapshot repositories
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 3]
```diff
@@ -24,6 +24,56 @@ Other versions:
...
+
+While there are many third-party storage systems, not all of them work well as
+snapshot repositories. Some systems perform poorly or behave incorrectly as a
+repository, especially when multiple {es} nodes access them concurrently.
+
+To avoid these problems, you can now use the
+{ref}/repo-analysis-api.html[repository analysis API] to test a repository's
```
Comment (DaveCTurner) [inline docs/reference/release-notes/highlights.asciidoc:71]:
```suggestion
{ref}/repo-analysis-api.html[repository analysis API] to analyze a repository's
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 4]
```diff
@@ -24,6 +24,56 @@ Other versions:
...
+
+[discrete]
+[[test-snapshot-repositories]]
+=== Test snapshot repositories
+
+While there are many third-party storage systems, not all of them work well as
+snapshot repositories. Some systems perform poorly or behave incorrectly as a
+repository, especially when multiple {es} nodes access them concurrently.
+
+To avoid these problems, you can now use the
```
Comment (DaveCTurner) [inline docs/reference/release-notes/highlights.asciidoc:70]: Optional suggestion:

```suggestion
To detect and avoid these problems, you can now use the
```

**Review by DaveCTurner — COMMENTED** [review]
I'd rather we stuck with the "analyze" terminology than "test". The trouble with "test" is that it implies that a repository which passes the test is ok to use, but that's not the right conclusion, it merely means we didn't find any problems.

**Comment by jrodewig** [comment]
Thanks for the feedback @DaveCTurner. I've incorporated your suggestions.

**Review by DaveCTurner — APPROVED** [review]
LGTM, thanks James

**Comment by jrodewig** [comment]
I'm going to go ahead and merge this in. @dakrone if you have any feedback, please feel free to leave a comment or suggestion here. I can either push the resulting changes or open up a follow-up PR. Thanks!

**Pull request merged**